packages/scandipwa/src/util/Request/Request.js

```diff
@@ -266,3 +266,26 @@ export const debounce = (callback, delay) => {
         timeout = setTimeout(() => callback.apply(context, args), delay);
     };
 };
+
+/** @namespace Util/Request */
+export class Debouncer {
+    timeout;
+
+    handler = () => {};
+
+    startDebounce = (callback, delay) => (...args) => {
+        const context = this;
+        clearTimeout(this.timeout);
+        this.handler = () => callback.apply(context, args);
+        this.timeout = setTimeout(this.handler, delay);
+    };
+
+    cancelDebounce = () => {
+        clearTimeout(this.timeout);
+    };
+
+    cancelDebounceAndExecuteImmediately = () => {
+        clearTimeout(this.timeout);
+        this.handler();
+    };
+}
```

Comment:
Basically it is same debounce that we already have but in class. If we need to use it like that, then better to change it everywhere (if it really need to be implemented like that). It is used to add debounce for async arrow functions? If yes, then better chose to actually remove arrows function. (it will be also possible to extend it on projects which is better choice)